### PR TITLE
Implement daml install --install-assistant flag.

### PR DESCRIPTION
**Pull request by associahedron**
- Renamed `--activate` to `--install-assistant` because the former is far too vague.
    - Deprecated and hid `--activate` flag -- it's still supported but will be removed at a later date.
- Changed `--install-assistant` to take on three values (`yes, no, auto`) like other flags, with `auto` being the default.
- Changed the default behavior to be "install the assistant whenever there is a newer version", just like auto-install. This way users almost never have to manage the assistant version explicitly, but still have the option of doing so.

### CODE REVIEW

File: daml-assistant/src/DAML/Assistant/Types.hs [hunk 1]
```diff
@@ -82,7 +84,8 @@ data VersionOptions = VersionOptions
 -- | Command-line options for daml install command.
 data InstallOptions = InstallOptions
     { iTargetM :: Maybe RawInstallTarget -- ^ version to install
-    , iActivate :: ActivateInstall -- ^ activate the assistant
+    , iAssistant :: InstallAssistant -- ^ install the assistant
+    , iActivate :: ActivateInstall -- ^ install the assistant if true (deprecated, delete with 0.14.x)
```
Comment (cocreature) [inline daml-assistant/src/DAML/Assistant/Types.hs]: I’d probably handled the logic for `--activate` in the parser and then only have `iAssistant` here but I don’t feel strongly about this.

I’m also not entirely sure we need backwards compatibility here but the added complexity seems small enough that I’m definitely not opposed to having it.

Comment (associahedron) [reply daml-assistant/src/DAML/Assistant/Types.hs]: I think I still need it here for the `--activate` deprecation warning. 

Yeah ... getting rid of `iActivate` and `--activate` is definitely something I want to do, perhaps in `0.14.x`. I just think that now that we have users we need to be careful not break user interfaces too quickly.

File: docs/source/support/release-notes.rst [hunk 2]
```diff
@@ -9,6 +9,13 @@ This page contains release notes for the SDK.
 HEAD — ongoing
 --------------
 
+DAML Assistant
+~~~~~~~~~~~~~~
+
+- Added `--install-assistant` flag to `daml install` command, changing the default
```
Comment (bethaitman) [inline docs/source/support/release-notes.rst]: sorry, rst
```suggestion
- Added ``--install-assistant`` flag to ``daml install`` command, changing the default
```

File: docs/source/support/release-notes.rst [hunk 3]
```diff
@@ -9,6 +9,13 @@ This page contains release notes for the SDK.
...
+  behavior of `daml install` to be "install the assistant whenever we are installing
```
Comment (bethaitman) [inline docs/source/support/release-notes.rst]:
```suggestion
  behavior of ``daml install`` to be "install the assistant whenever we are installing
```

File: docs/source/support/release-notes.rst [hunk 4]
```diff
@@ -9,6 +9,13 @@ This page contains release notes for the SDK.
...
+  a newer version of the SDK". Deprecated the `--activate` flag.
```
Comment (bethaitman) [inline docs/source/support/release-notes.rst]:
```suggestion
  a newer version of the SDK". Deprecated the ``--activate`` flag.
```

File: docs/source/support/release-notes.rst (hunk 2 above)
Comment (associahedron) [reply docs/source/support/release-notes.rst]: Oops, thanks! I fixed these up in #1786.

**Pull request merged**